fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/KeySpacePathImportDataTest.java

```diff
@@ -0,0 +1,419 @@
+
+        copyData(keySpace.root(), keySpace.root());
+
+        verifySingleKey(dataStore, Tuple.from("record2", 0), Tuple.from("user100_app1_data2_0"));
+    }
+
+    @Test
+    void importDataWithDuplicateKeys() {
+        // Test importing data where the same key appears multiple times in the input
+        // Should verify that the last value wins for duplicate keys
+        final String rootUuid = UUID.randomUUID().toString();
+        KeySpace root = new KeySpace(
+                new KeySpaceDirectory("root", KeyType.STRING, rootUuid)
+                        .addSubdirectory(new KeySpaceDirectory("data", KeyType.LONG)));
+
+
+        KeySpacePath dataPath = root.path("root").add("data", 1L);
+
+        // Create multiple DataInKeySpacePath objects with same key but different values
+        List<DataInKeySpacePath> duplicateData = Arrays.asList(
+                new DataInKeySpacePath(dataPath, Tuple.from("item"), Tuple.from("first_value").pack()),
+                new DataInKeySpacePath(dataPath, Tuple.from("item"), Tuple.from("second_value").pack()),
+                new DataInKeySpacePath(dataPath, Tuple.from("item"), Tuple.from("final_value").pack())
+        );
+
+        try (FDBRecordContext context = database.openContext()) {
+            root.path("root").importData(context, duplicateData).join();
+            context.commit();
+        }
+
+        verifySingleKey(dataPath, Tuple.from("item"), Tuple.from("final_value"));
+    }
+
+    private void setSingleKey(KeySpacePath path, Tuple remainder, Tuple value) {
+        try (FDBRecordContext context = database.openContext()) {
+            byte[] key = path.toSubspace(context).pack(remainder);
+            context.ensureActive().set(key, value.pack());
+            context.commit();
+        }
+    }
+
+    private void verifySingleKey(KeySpacePath path, Tuple remainder, Tuple expected) {
+        try (FDBRecordContext context = database.openContext()) {
+            byte[] key = path.toSubspace(context).pack(remainder);
+            assertEquals(expected, Tuple.fromBytes(context.ensureActive().get(key).join()));
+        }
+    }
+
+    private static void setInPath(final KeySpacePath path, final FDBRecordContext context,
+                                  final Tuple remainder, final Object value) {
+        byte[] key = path.toSubspace(context).pack(remainder);
+        context.ensureActive().set(key, Tuple.from(value).pack());
+    }
+
+    private static Tuple getTupleFromPath(final FDBRecordContext context, final KeySpacePath path, final Tuple remainder) {
+        byte[] key = path.toSubspace(context).pack(remainder);
+        return Tuple.fromBytes(context.ensureActive().get(key).join());
+    }
+
+    private void copyData(final KeySpacePath sourcePath, KeySpacePath destinationPath) {
+        // Export the data
+        final List<DataInKeySpacePath> exportedData = getExportedData(sourcePath);
+
+        if (databases.size() > 1) {
```

Comment:
I feel this allowance for any configuration may introduce confusion and inconsistency between the tests that are actually running in each environment. Wouldn't it be more predictable if there were specific sets of tests targeting single- and double- DB scenarios?